### PR TITLE
Update: Add an ignoreRegexpLiterals option to max-len (fixes #3229)

### LINKED ISSUE #3229: max-len to ignore Regular Expressions
Is there a way to add an option to ignore regular expressions that are too long? Sometimes those things can get huge. For example, URI.js needs a very long regex to parse a URL: https://github.com/medialize/URI.js/blob/gh-pages/src/URI.js#L207

### PR DESCRIPTION
**Pull request by Wilfred**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Fixes issue #3229.

**Please check each item to ensure your pull request is ready:**

- [x] I've read the [pull request guide](http://eslint.org/docs/developer-guide/contributing/pull-requests)
- [x] I've included tests for my change
- [x] I've updated documentation for my change (if appropriate)

**What changes did you make? (Give an overview)**

Added an `ignoreRegexpsLiteral` option, added tests, and updated docs.

**Is there anything you'd like reviewers to focus on?**

Any code style issues: this is my first time working on the eslint codebase.

### CODE REVIEW

**Comment by platinumazure** [comment]
Labeling accepted/enhancement/rule to match the referenced issue.

**Comment by ljharb** [comment]
@not-an-aardvark that's a good point - probably.

File: docs/rules/max-len.md [hunk 1]
```diff
@@ -131,6 +132,16 @@ Examples of **correct** code for this rule with the `{ "ignoreTemplateLiterals":
 var longTemplateLiteral = `this is a really really really really really long template literal!`;
 ```
 
+### ignoreRegexpLiterals
```
Comment (kaicataldo) [inline docs/rules/max-len.md]: Agreed with @not-an-aardvark, I think this should be `ignoreRegExpLiterals`

**Comment by Wilfred** [comment]
OK, I've renamed the option.

**Comment by kaicataldo** [comment]
Thanks!

File: docs/rules/max-len.md [hunk 2]
```diff
@@ -23,6 +23,7 @@ This rule has a number or object option:
 * `"ignoreUrls": true` ignores lines that contain a URL
 * `"ignoreStrings": true` ignores lines that contain a double-quoted or single-quoted string
 * `"ignoreTemplateLiterals": true` ignores lines that contain a template literal
+* `"ignoreRegExpLiterals": true` ignores lines that contain a regexp literal
```
Comment (not-an-aardvark) [inline docs/rules/max-len.md]: Replace the end of this line with `contain a RegExp literal`

File: lib/rules/max-len.js [hunk 3]
```diff
@@ -227,6 +231,18 @@ module.exports = {
 
 
         /**
+         * Retrieves an array containing all regexp literals in the source code.
+         *
+         * @returns {ASTNode[]} An array of regexp literal nodes.
+         */
+        function getAllRegexpLiterals() {
```
Comment (not-an-aardvark) [inline lib/rules/max-len.js]: Same thing here

File: docs/rules/max-len.md [hunk 4]
```diff
@@ -131,6 +132,16 @@ Examples of **correct** code for this rule with the `{ "ignoreTemplateLiterals":
...
 ```
 
+### ignoreRegExpLiterals
+
+Examples of **correct** code for this rule with the `{ "ignoreRegExpLiterals": true }` option:
+
+```js
+/*eslint max-len: ["error", { "ignoreRegExpLiterals": true }]*/
+
+var longRegexpLiteral = /this is a really really really really really long regular expression!/;
```
Comment (not-an-aardvark) [inline docs/rules/max-len.md]: For consistency, could you make this `longRegExpLiteral` as well?

**Comment by Wilfred** [comment]
OK, I've updated the PR with the changes requested.

**Comment by btmills** [comment]
Thanks @Wilfred!

**Pull request merged**